src/Microsoft.Extensions.TelemetryAdapter/Internal/ProxyMethodEmitter.cs

```diff
@@ -28,7 +29,9 @@ public static class ProxyMethodEmitter
             var inputTypeInfo = inputType.GetTypeInfo();
             for (var i = 0; i < parameters.Length; i++)
             {
-                var property = inputTypeInfo.GetDeclaredProperty(parameters[i].Name);
+                var property = inputTypeInfo.DeclaredProperties
+                    .Where(p => string.Equals(p.Name, parameters[i].Name, StringComparison.OrdinalIgnoreCase))
+                    .FirstOrDefault();
```

Comment:
I debated whether or not to bother. Might as well just so there's never a question about it :+1: